### PR TITLE
USB: Fix file writing hang regression for big files

### LINKED ISSUE #3452: USB: File writing hang regression for big files
### Describe the bug.

Have 10+ user reports, including first hand, of qFlipper consistently hanging during the update process while writing resources.tar (although at random points throughout the write operation). I tested back through the commit history and narrowed it down to #3358. In particular, reverting the USB CDC parameter changes fixes the issues.

### Reproduction

1. Connect to qFlipper
2. Start an update operation (eg. Install from file)
3. At some random point through resources.tar, writes hang
4. qFlipper reports timeout, update fails

### Target

Flipper USB -> qFlipper on all operating systems

### Logs

In Flipper serial debug logs, write operations flow through normally until they suddenly stop, for no apparent reason, and a few seconds later qFlipper reports a timeout error. No other information is present in the logs.

### Anything else?

The PR in question mentions "switch from 2x unidirectional data endpoints to 1x bidirectional".
Seems like this is what is causing issues, although not clear why. Is there any more info on this change?

### PR DESCRIPTION
**Pull request by WillyJL**
# What's new

- Fixes #3452 as it is still an issue
- TLDR:
  - #3358 fixed crashes when using `cat` on flipper serial
  - it also included some unrelated USB CDC parameter changes
  - from what I've tested, the crash for `cat` on serial is fixed even when not including the USB CDC changes
  - this additional unrelated change was only denoted as `USB CDC: switch from 2x unidirectional data endpoints to 1x bidirectional`
  - from what I can see there are no additional details, no bugs this is attempting to fix, and no improvements this brings
  - there are however some very real issues with this tiny USB CDC changes
  - during USB file writing (like qFlipper or Web update) there is a random chance of USB lockup
  - when this happens, Flipper USB will be fully unresponsive until rebooted
  - qFlipper will say timeout after some seconds, Web will just remain stuck forever
  - the bigger the file being written over USB, the higher chance of lockup
  - it is most prominent when installing some big CFW (6MB+ resources.tar), but I have seen multiple users struggling to update OFW due to this bug
  - in other cases I told people to downgrade to older OFW to avoid this bug, and they could not even downgrade due to having newer OFW currently installed with this bug
  - CFWs have reverted this changes in same way as this PR and have had no issues since. going from CFW to new CFW, or from CFW to OFW
  - but since OFW after 0.99 has this bug, if user has OFW installed they can have issues updating to newer OFW or installing CFW

# Verification 

- Install firmware with big resources.tar multiple times and no issues

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Comment by skotopes** [comment]
https://github.com/flipperdevices/flipperzero-firmware/pull/3358 actually fixes two problems: 
- dual endpoint configuration causes linux to send garbage to the device
- line buffer overflow in cli caused by garbage sent to the device

So what you doing here is breaking first one. I'd rather see proper fix that is finding root cause of single endpoint cdc crash.

**Comment by WillyJL** [comment]
I see, thanks for clearing that up. It's the crucial piece of information that wasn't specified anywhere (atleast as far as I could see).

I'll try to find out more but it is really difficult to debug, there are no weird logs or crashes usually, just stops working. It almost feels like random memory corruption.

**Comment by skotopes** [comment]
Correct fix that properly covers all cases in #3705
